packages/@headlessui-react/src/hooks/use-element-size.ts

```diff
@@ -12,15 +12,19 @@ export function useElementSize(
   unit = false
 ) {
   let element = ref === null ? null : 'current' in ref ? ref.current : ref
-  let [size, setSize] = useState(() => computeSize(element))
+  let [identity, forceRerender] = useReducer(() => ({}), {})
```

Comment:
The new object will always be a new instance and `===` won't match anymore. Therefore we can use this empty object as a stable identifier to know whether we should re-render or not.